src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { Request, Response } from 'express';
+import UserService from '../services/user.service';
+import { HttpError } from '../helpers/error-handler';
+import { userServiceInstance } from './../services/user.service';
+import * as jwt from 'jsonwebtoken';
+
+const accessTokenSecret = 'badwayofsetingaccesstoken';
+
+export default class AuthController {
+
+  constructor(public userService: UserService) {}
+
+  login (req: Request, res: Response, next): void {
+    const { username, password } = req.body;
+
+    this.userService.getUsers({loginSubstring: username, limit:1, isFullLogin:true})
+        .then( users => {
+          if (users[0].login === username && users[0].password === password) {
+            const accessToken = jwt.sign({ username: users[0].login}, accessTokenSecret, { expiresIn: '1h' });
+
+            res.json({
+              accessToken
+            });
+          } else {
+            throw new HttpError(404,'Username of password is incorrect');
+          }
+        }).catch(error => {
+          next(error);
+        });
+  }
+
+  checkJWT(req: Request, res: Response, next): void {
```

Comment:
isn't is a middleware, is it? If so, it can be moved accordingly from controller